### PR TITLE
API: Fix splatalogue module with upstream changes

### LINKED ISSUE #2956: New Splatalogue breaks astroquery interface
Example:
```python
>>> Splatalogue.query_lines(1*u.GHz, 2*u.GHz)
Traceback (most recent call last):
  Cell In[239], line 1
    Splatalogue.query_lines(1*u.GHz, 2*u.GHz)
  File ~/repos/astroquery/astroquery/utils/class_or_instance.py:25 in f
    return self.fn(obj, *args, **kwds)
  File ~/repos/astroquery/astroquery/utils/process_asyncs.py:29 in newmethod
    result = self._parse_result(response, verbose=verbose)
  File ~/repos/astroquery/astroquery/splatalogue/core.py:480 in _parse_result
    result = ascii.read(response.text.split('\n'), delimiter=':',
  File ~/mambaforge/envs/py39forge/lib/python3.9/site-packages/astropy/io/ascii/ui.py:426 in read
    dat = reader.read(table)
  File ~/mambaforge/envs/py39forge/lib/python3.9/site-packages/astropy/io/ascii/core.py:1451 in read
    raise InconsistentTableError(errmsg)
InconsistentTableError: Number of header columns (1) inconsistent with data columns (2) at data line 13
Header values: ['<!-- start of nrao-header.html -->']
Data values: ['<a href="http', '//www.nrao.edu/">']
```

Announced and deployed the same day:
https://science.nrao.edu/enews/17.2/

This is a severely breaking change, such that the existing code is likely now of zero value.  

I used splatalogue a lot, so I may want to write a new module if the upstream interface comes back online (presently, the new splatalogue does not function at all), but in the meantime I recommend people switch to the JPL and CDMS line lists.

### PR DESCRIPTION
**Pull request by keflavich**
This is a major refactor that will fix splatalogue once I've found all the corners.

Pushing now so I have a WIP saved here.

EDIT: Fixes #2956

### CODE REVIEW

**Comment by keflavich** [comment]
This works locally.  I was even able to reproduce the error that is showing up now about the wrong content type in the mocks, and fix it, but the same error is recurring now.  I can't make sense of it now.  The `content` keyword in the mock is very clearly a `b''` that has a `decode` attribute.

**Comment by keflavich** [comment]
ach, nevermind my previous comment; I didn't realize where the mocks were defined (they're in `conftest`)

**Comment by keflavich** [comment]
OK, looks like we're green.  The readthedocs failure doesn't seem to be associated with this PR, as far as I can see.

**Comment by keflavich** [comment]
@bsipocz this is ready for your review.  Any ideas on the readthedocs issue?

**Comment by bsipocz** [comment]
RTD failure is pyvo related and I'll look into it separately.

**Comment by keflavich** [comment]
@bsipocz is there anything holding this back from getting merged?

**Comment by bsipocz** [comment]
> @bsipocz is there anything holding this back from getting merged?

Sorry, apparently I have a half finished review here with some minor comments. RTD has been fixed, so that is not holding it up.

**Comment by keflavich** [comment]
Do I need to rebase or rerun tests to get RTD to pass?

**Review by bsipocz — CHANGES_REQUESTED** [review]
There are multiple failures in the RST docs, those needs to be fixed (most of them are cosmetic, but the last ones seem genuine due to the changed column names).

The rest of the inline comments are minor and would be happy to see them addressed, but it's also OK if they are not.

**Comment by bsipocz** [comment]
> Do I need to rebase or rerun tests to get RTD to pass?

No need to rebase as not much has changed overall, but also wouldn't hurt once you fixed the docs.

File: astroquery/splatalogue/core.py [hunk 1]
```diff
@@ -292,135 +274,170 @@ def _parse_kwargs(self, *, min_frequency=None, max_frequency=None,
...
-
-        if band != 'any':
-            if band not in self.FREQUENCY_BANDS:
-                raise ValueError("Invalid frequency band.")
-            if min_frequency is not None or max_frequency is not None:
-                warnings.warn("Band was specified, so the frequency "
-                              "specification is overridden")
-            payload['band'] = band
-        elif min_frequency is not None and max_frequency is not None:
+        payload = {"searchSpecies": "",
```
[the remark this answers is not in the document]
Comment (keflavich) [reply astroquery/splatalogue/core.py:277]: Not as far as I'm aware.  It gets populated by a javascript script `main.js` with this function (below) that we could back out parameters from, but it's not worth my time to figure that out:
```js
                getDataDict() {
                    var t = this.advancedLineIntesity
                      , i = 0;
                    return this.advancedLineIntesity.startsWith("CDMS/JPL (log)") ? i = this.advancedLineIntesityCDMS : this.advancedLineIntesity.startsWith("Sij-mu2") ? i = this.advancedLineIntesitySijmu2 : this.advancedLineIntesity.startsWith("Aij (log)") && (i = this.advancedLineIntesityAij),
                    {
                        searchSpecies: this.searchspeciesinput,
                        speciesSelectBox: this.speciesSelectBoxIds,
                        dataVersion: this.advancedDataVersion,
                        userInputFrequenciesFrom: this.FreqRangesFrom,
                        userInputFrequenciesTo: this.FreqRangesTo,
                        userInputFrequenciesUnit: this.addvancedFreqUnit,
                        frequencyRedshift: this.redshiftvalue,
                        energyFrom: this.advancedEnergyFrom,
                        energyTo: this.advancedEnergyTo,
                        energyRangeType: this.advancedEnergyType,
                        lineIntensity: t,
                        lineIntensityLowerLimit: i,
                        excludeAtmosSpecies: this.advancedExAtmosSpecies,
                        excludePotentialInterstellarSpecies: this.advancedExPotInterSpecies,
                        excludeProbableInterstellarSpecies: this.advancedExProbInterSpecies,
                        excludeKnownASTSpecies: this.advancedExASTSpecies,
                        showOnlyAstronomicallyObservedTransitions: this.advancedOnlyObTransition,
                        showOnlyNRAORecommendedFrequencies: this.advancedOnlyNRAORecd,
                        lineListDisplayJPL: this.advancedJPL,
                        lineListDisplayCDMS: this.advancedCDMS,
                        lineListDisplayLovasNIST: this.advancedLoveNIST,
                        lineListDisplaySLAIM: this.advancedSLAIM,
                        lineListDisplayToyaMA: this.advancedToyaMA,
                        lineListDisplayOSU: this.advancedOSU,
                        lineListDisplayRecombination: this.advancedRecomb,
                        lineListDisplayTopModel: this.advancedTopModel,
                        lineListDisplayRFI: this.advancedRFI,
                        lineStrengthDisplayCDMSJPL: this.advancedStrCDMS,
                        lineStrengthDisplaySijMu2: this.advancedStrSimjMu2,
                        lineStrengthDisplaySij: this.advancedStrSij,
                        lineStrengthDisplayAij: this.advancedStrAij,
                        lineStrengthDisplayLovasAST: this.advancedStrLovasAST,
                        energyLevelOne: this.advancedEngEcm1,
                        energyLevelTwo: this.advancedEngEK,
                        energyLevelThree: this.advancedEngEUpcm,
                        energyLevelFour: this.advancedEngEUpK,
                        displayObservedTransitions: this.advancedDispObsTrans,
                        displayG358MaserTransitions: this.advancedDispG358,
                        displayObservationReference: this.advancedDispObsRef,
                        displayObservationSource: this.advancedDispObsSrc,
                        displayTelescopeLovasNIST: this.advancedDispTelLovas,
                        frequencyErrorLimit: this.advancedNoFreqError,
                        displayHFSIntensity: this.advancedDispHFS,
                        displayUnresolvedQuantumNumbers: this.advancedDispUnResQuat,
                        displayUpperStateDegeneracy: this.advancedDispUpSteDegn,
                        displayMoleculeTag: this.advancedDispMoleTag,
                        displayQuantumNumberCode: this.advancedDispQuatCode,
                        displayLabRef: this.advancedDispLabRef,
                        displayOrderedFrequencyOnly: this.advancedDispOrdFreqOnly,
                        displayNRAORecommendedFrequencies: this.advancedDispNRAORecFreq,
                        displayUniqueSpeciesTag: this.advancedDispUnqSpeciesTag,
                        displayUniqueLineIDNumber: this.advancedDispUnqLineID,
                        exportType: this.advancedExportType,
                        exportDelimiter: this.advancedExportDelimiter,
                        exportLimit: this.advancedExportRange,
                        exportStart: this.advancedExportStart,
                        exportStop: this.advancedExportStop
                    }
                }
```

File: astroquery/splatalogue/core.py [hunk 2]
```diff
@@ -292,135 +274,170 @@ def _parse_kwargs(self, *, min_frequency=None, max_frequency=None,
...
 
-        if top20 is not None:
-            if top20 in self.TOP20_LIST:
-                payload['top20'] = top20
-            else:
-                raise ValueError("Top20 is not one of the allowed values")
-        elif chemical_name in ('', {}, (), [], set()):
-            # include all
-            payload['sid[]'] = []
+        if chemical_name in ('', {}, (), [], set(), None):
```
[the remark this answers is not in the document]
Comment (keflavich) [reply astroquery/splatalogue/core.py:345]: I think yes?  But I am worried about exactly the case you mention, i.e., `''` is passed in.

i could simplify this to `chemical_name in ('', None)`, though; the rest of the 'null' options don't really make sense.

**Comment by bsipocz** [comment]
RTD failure is now relevant to this PR.

File: docs/splatalogue/splatalogue.rst [hunk 3]
```diff
@@ -218,48 +218,37 @@ regular basis.  For example, if you want data returned preferentially in units
...
+           109  Formaldehyde                         HDCO              1(1,0) - 1(1,1)   5346.1416 -8.31616             11.18301
+           155  Formaldehyde H<sub>2</sub>C<sup>18</sup>O       1(  1, 0)-   1(  1, 1)    4388.797 -8.22052             15.30187
+           155  Formaldehyde H<sub>2</sub>C<sup>18</sup>O  1( 1, 0)- 1( 1, 1), F= 1- 0   4388.7783  -9.0498             15.30206
+           155  Formaldehyde H<sub>2</sub>C<sup>18</sup>O  1( 1, 0)- 1( 1, 1), F= 0- 1   4388.7957 -8.57268             15.30206
+           155  Formaldehyde H<sub>2</sub>C<sup>18</sup>O  1( 1, 0)- 1( 1, 1), F= 2- 2   4388.7965 -8.69765             15.30206
+           155  Formaldehyde H<sub>2</sub>C<sup>18</sup>O              1(1,0) - 1(1,1)    4388.797 -8.57272             15.30206
+           155  Formaldehyde H<sub>2</sub>C<sup>18</sup>O  1( 1, 0)- 1( 1, 1), F= 2- 1   4388.8012 -9.17475             15.30206
+           155  Formaldehyde H<sub>2</sub>C<sup>18</sup>O  1( 1, 0)- 1( 1, 1), F= 1- 2   4388.8036  -8.9529             15.30206
+
+There are utility functions in `astroquery.splatalogue.utils` that automate
```
Comment (bsipocz) [inline docs/splatalogue/splatalogue.rst:250]: These are not in the public API, should we 1) put it there 2) maybe no advertise it?

Either case, the fix below gets rid of the RTD CI failure
```suggestion
There are utility functions in ``astroquery.splatalogue.utils`` that automate
```

Comment (keflavich) [reply docs/splatalogue/splatalogue.rst]: They kinda should be in the public API, but I don't want to make that change now - your workaround is better for the moment.

**Review by bsipocz — APPROVED** [review]
I'll rebase this to fix the changelog conflict and then merge.

Thanks @keflavich!

**Pull request merged**